pkg/scheduler/scheduler.go

```diff
@@ -608,6 +612,16 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 		return
 	}
 
+	// At the end of a successful scheduling cycle, pop and move up Pods if needed.
+	if len(podsToMove.PodsMap) != 0 {
```

Comment:
This place does not need to acquire lock under the normal logic.
But a little doubt：Is it possible that some other users will start a new go-routine in their plugins to R/W `podsToMove`?

Reply:
> Is it possible that some other users will start a new go-routine in their plugins to R/W `podsToMove`?

I cannot think of a reason why users would do that. Doing that means you choose to delay a piece of important information in a random manner (as you start a goroutine).